qiskit/pulse/library/parametric_pulses.py

```diff
@@ -92,6 +93,35 @@ def parameters(self) -> Dict[str, Any]:
         """Return a dictionary containing the pulse's parameters."""
         pass
 
+    def assign_parameters(self,
+                          value_dict: Dict[ParameterExpression,
+                                           Union[ParameterExpression, int, float, complex]]
+                          ) -> 'ParametricPulse':
+        """Return a new ParametricPulse with parameters assigned.
+
+        Args:
+            value_dict: A mapping from Parameters to either numeric values or another
+                Parameter expression.
+
+        Returns:
+            New pulse with updated parameters.
+        """
+        if not any([is_parameterized(val) for val in self.parameters.values()]):
+            return self
+
+        new_parameters = {}
+        for op, op_value in self.parameters.items():
+            for parameter, value in value_dict.items():
+                if is_parameterized(op_value) and parameter in op_value.parameters:
+                    op_value = op_value.assign(parameter, value)
+                    try:
+                        # TODO: some need to be complex, but ParameterExpression doesn't support it
```

Comment:
This is a bummer. Typing is definitely needed in ParameterExpression.

Reply:
I mean ParameterExpressions error on complex values -- I think there might be work in progress to add it

Reply:
But you are also right about needing typing in ParameterExpression